### PR TITLE
Fix osx busy cursor

### LINKED ISSUE #9156: Saving .png figure failing on OSX backend
### Bug report
Saving `.png` files is currently failing on the osx backend. I'm guessing this is due to #6603 (see the error message below)

**Code for reproduction**

```python
import matplotlib
matplotlib.use('MacOSX')
import matplotlib.pyplot as plt

fig, ax = plt.subplots()
fig.savefig('test.png')
```

**Actual outcome**

```python
  File "test.py", line 6, in <module>
    fig.savefig('test.png')
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/figure.py", line 1813, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/backend_bases.py", line 2258, in print_figure
    **kwargs)
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/backends/backend_agg.py", line 507, in print_png
    FigureCanvasAgg.draw(self)
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/backends/backend_agg.py", line 429, in draw
    toolbar.set_cursor(cursors.WAIT)
  File "/Users/dstansby/github/matplotlib/lib/matplotlib/backends/backend_macosx.py", line 188, in set_cursor
    _macosx.set_cursor(cursor)
SystemError: <built-in function set_cursor> returned NULL without setting an error
```

**Expected outcome**

**Matplotlib version**

  * Operating System: OSX
  * Matplotlib Version: Current master branch installed using pip
  * Python Version: 3.6.2

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #9156. See https://github.com/matplotlib/matplotlib/issues/9156#issuecomment-327217962 for more information on the fix.

### CODE REVIEW

**Comment by anntzer** [comment]
LGTM although I can't actually test it :-) Also, does OSX actually display a busy spinner in that case?  (Not that I'd consider that a blocker)

**Comment by dstansby** [comment]
Oh yeah, should have added that this fixes the issue for me! I think one other osx dev should check this before it goes in. It doesn't look like OSX is showing a spinner when I plot and move a very large scatter plot.

**Comment by dopplershift** [comment]
What about fixing the problem that it returns null without an exception set?

**Comment by anntzer** [comment]
It should finely be fixed at some point but I don't think it's release critical, considering that the issue has always been present.

**Comment by dopplershift** [comment]
Fair enough.

File: src/_macosx.m [hunk 1]
```diff
@@ -1902,6 +1902,7 @@ -(void)save_figure:(id)sender
       case 1: [[NSCursor arrowCursor] set]; break;
       case 2: [[NSCursor crosshairCursor] set]; break;
       case 3: [[NSCursor openHandCursor] set]; break;
+      case 4: break;
```
Comment (dopplershift) [inline src/_macosx.m]: Can we get a comment here about the fact that OS X handles the busy for us and doesn't provide an icon?

**Pull request merged**